src/main/java/mort/parser/Parser.java

```diff
@@ -0,0 +1,210 @@
+package mort.parser;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import mort.command.AddCommand;
+import mort.command.Command;
+import mort.command.CommandWord;
+import mort.command.DeleteCommand;
+import mort.command.ExitCommand;
+import mort.command.FindCommand;
+import mort.command.ListCommand;
+import mort.command.MarkCommand;
+import mort.command.UnmarkCommand;
+import mort.exception.MortException;
+import mort.task.TaskType;
+import mort.ui.Ui;
+
+/**
+ * Parser for parsing user inputs, dates and tasks.
+ */
+public class Parser {
+    /**
+     * Parses and validates user input and returns a command to be executed.
+     * @param fullCommand The command input by the user.
+     * @return The Command to be executed.
+     * @throws MortException If command is invalid.
+     */
+    public static Command parse(String fullCommand) throws MortException {
+        String strippedCommand = fullCommand.stripTrailing();
+        String[] parsedCommand = strippedCommand.split(" ", 2);
+        try {
+            switch (parsedCommand[0]) {
```

Comment:
Really good use of the switch statement here due to the numerous cases involved!